docs/next/guides/building-and-testing/plugins.md

```diff
@@ -1,242 +1,175 @@
+    // Reset all of your plugin class properties to their default values here.
+    this.configuration = null;
 
     // The super method takes care of clearing the hook connections and assigning the 'false' value to the 'this.enabled' property.
     super.disablePlugin();
   }
 
   /**
-   * The updatePlugin method is called on the afterUpdateSettings hook (unless the updateSettings method turned the plugin off).
-   * It should contain all the stuff your plugin needs to do to work properly after the Handsontable instance settings were modified.
+   * The updatePlugin method is called on the `afterUpdateSettings` hook
+   * (unless the updateSettings method turned the plugin off).
+   *
+   * It should contain all the stuff your plugin needs to do to work correctly
+   * after updating Handsontable instance settings.
    */
   updatePlugin() {
-
-    // The updatePlugin method needs to contain all the code needed to properly re-enable the plugin. In most cases simply disabling and enabling the plugin should do the trick.
+    // The `updatePlugin` method needs to contain all the code needed to properly re-enable the plugin.
+    // In most cases simply disabling and enabling the plugin should do the trick.
     this.disablePlugin();
     this.enablePlugin();
 
     super.updatePlugin();
   }
 
+  /**
+   * Define your external methods.
+   */
+  externalMethodExample() {
+    // Method definition.
+  }
+
   /**
    * The afterChange hook callback.
    *
-   * @param {Array} changes Array of changes.
-   * @param {String} source Describes the source of the change.
+   * @param {CellChange[]} changes An array of changes.
+   * @param {string} source Describes the source of the change.
    */
   onAfterChange(changes, source) {
     // afterChange callback goes here.
   }
 
   /**
-   * The destroy method should de-assign all of your properties.
+   * The `destroy` method is the best place to clean up any instances,
+   * objects or index mappers created during the plugin's lifecycle.
    */
   destroy() {
-    // The super method takes care of de-assigning the event callbacks, plugin hooks and clearing all the plugin properties.
+    // The super method takes care of de-assigning the event callbacks,
+    // plugin hooks and clearing all the plugin properties.
     super.destroy();
   }
 }
-
-export {InternalPluginSkeleton};
-
-// You need to register your plugin in order to use it within Handsontable.
-registerPlugin('internalPluginSkeleton', InternalPluginSkeleton);
-
 ```
 
-#### External plugin
+### 3. Register CustomPlugin
+Last but not least, you need to register the plugin. This is the only way to use plugins in Handsontable.
+There are two options.
 
-Similarly to the above, this template is [hosted on GitHub](https://github.com/handsontable/handsontable-skeleton/tree/master/plugins/external).
+- You can define a static getter named `PLUGIN_KEY` that the `registerPlugin` utility will use as the plugin's alias.
```

Comment:
Something like "As we did in the example before" would assure the user that the getter was a good idea and now he should choose this option.